### ISSUE TITLE
[Bug] [1.9.0-2b]: Search won't work.

### ISSUE THREAD

**Issue by WinnerWind** [KosmosisDire/obsidian-webpage-export#505]
### What happened?

When exporting the website and hosting it on local host using `python3 -m http.server` (Runs the current directory as an Apache server), I expect the search function to work, or show a window, or give some response. No response is seen, and the search bar just works as a default textbar

### Reproduction

1. Clear Cache and Purge (to be on the safe side)
2. Export site with default settings
3. Host site
4. Search bar exists, so try to type something.
5. No response

### Last Working Version

Unsure

### Version

1.9.0-2b

### Operating System

Arch Linux x86_64 using X11

### Obsidian Version

current v1.6.7    installer v1.6.7

### Which browsers can you reproduce this in?

Firefox

### Relevant log output

```yaml
Button won't work. Sorry!
```

**Comment by KosmosisDire** [comment]
Thanks, I believe the search should be working in the beta and I am not able to reproduce. Can you check if there is console output on the website itself (in the browser) when you try to use the search?

**Comment by WinnerWind** [comment]
There seems to be no relevant log output.
![image](https://github.com/user-attachments/assets/62924f63-0f30-43f1-9965-8a031d8d3036)
I pressed enter even after entering the search and no errors were seen. I scrolled to the bottom just to be safe.

**Comment by KosmosisDire** [comment]
Huh, very strange. I am not able to reproduce this at all (I even used python to host a local server like you did). Are you able to reproduce this with a basic sandbox vault?

Would it be possible for you to share a zip of your exported vault (the exported site not the vault itself) with me so I can try and figure out what is wrong?

Thanks!

**Comment by WinnerWind** [comment]
Hi there! I'm not at home right now, but I'll do so when I'm back. I'll try reproducing with a sandbox vault, and sending the exported files.

**Comment by WinnerWind** [comment]
Closing this issue. Managed to find the cause, but it's a rather counter intuitive one.

I use Linux, and there was a recent update to the kernel. This broke a lot of my sensitive programs, including the program I used to clean up files. I noticed that that program kept trying to clean up the search index for the website. Well, it works now, sorry for taking up your time!

**Comment by KosmosisDire** [comment]
No worries! I'm glad you found the cause and it's working now! :)

**Comment by WinnerWind** [comment]
Well, search just broke again, and no its not the script I mentioned.
I guess I'll leave it be, since this IS a beta version. Eventually it might just fix itself.

**Comment by KosmosisDire** [comment]
So I was able to test this with a site I found which had this error even though I can't reproduce. I believe it is a problem with having an outdated version of mini search (at least in this case). You could try finding the latest minified mini search version and copy paste it into the mini search js script in the site lib folder.

Does it only change after re-exporting?

**Comment by WinnerWind** [comment]
Hi there! Sorry for the late reply.
I'll find the Minisearch you are talking about. You mean [this one](https://lucaong.github.io/minisearch/) right?

Does it change after re-exporting? No. The behaviour seems to be inconsistent. Sometimes search works after a full purge then re-export, sometimes it doesn't. I am not sure what is going on.

**Comment by KosmosisDire** [comment]
> Hi there! Sorry for the late reply. I'll find the Minisearch you are talking about. You mean [this one](https://lucaong.github.io/minisearch/) right?
> 
> Does it change after re-exporting? No. The behaviour seems to be inconsistent. Sometimes search works after a full purge then re-export, sometimes it doesn't. I am not sure what is going on.
> 
> Edit: Whenever exporting, the plugin always warns me that there is no search index and it will make a new one. Not sure if that's a behaviour of "Clear Cache" which I always do to rexport the entire vault.

Yes the missing search index shows up because of the full purge. 
https://cdn.jsdelivr.net/npm/minisearch@7.1.0/dist/umd/index.min.js
This is the minisearch cdn. Try pasting it's content into the minisearch.js file in the `site-lib/scripts/minisearch.js` file in your export and then reloading your site (making sure to clear the cache)

**Comment by WinnerWind** [comment]
Hi there! The fixed Minisearch does work. I did not need to reload the cache. I jus tpasted the file, refreshed the site, and for good measure, started the python hosting script again, and it works.
![image](https://github.com/user-attachments/assets/4736a0a3-adff-47cb-88aa-d507670231f9)
![image](https://github.com/user-attachments/assets/07407d15-1ce8-4ffe-afe8-0c03ef0967aa)
![image](https://github.com/user-attachments/assets/57b14309-4e2b-4c47-a4d8-83cab8348fda)

**Comment by KosmosisDire** [comment]
Yay! Great! I have updated the minisearch version which will be out in the next release

**Comment by WinnerWind** [comment]
Looking forward to the next release! Thanks for your dedication.
